staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -333,6 +345,254 @@ func TestApfCancelWaitRequest(t *testing.T) {
+// the specified desired concurrency share = concurrency limit
+func getServerConcurrency(desiredShare int) int {
+	catchAllShare := fcboot.MandatoryPriorityLevelConfigurationCatchAll.Spec.Limited.AssuredConcurrencyShares
+	return int(catchAllShare) + desiredShare
+}
+
+// returns a started http2 server, with a client function to send request to the server.
+func newHTTP2ServerWithClient(handler http.Handler) (*httptest.Server, func(path string) (*http.Response, error)) {
+	server := httptest.NewUnstartedServer(handler)
+	server.EnableHTTP2 = true
+	server.StartTLS()
+
+	return server, func(path string) (*http.Response, error) {
+		return server.Client().Get(server.URL + path)
+	}
+}
+
+// verifies that the expected flow schema and priority level UIDs are attached to the header.
+func expectMatchingAPFHeaders(t *testing.T, w http.ResponseWriter, expectedFS, expectedPL string) {
+	if w == nil {
+		t.Fatal("expected a non nil HTTP response")
+	}
+
+	key := flowcontrol.ResponseHeaderMatchedFlowSchemaUID
+	if value := w.Header().Get(key); expectedFS != value {
+		t.Fatalf("expected an HTTP header: %s=%s, but got %s=%s", key, expectedFS, key, value)
+	}
+
+	key = flowcontrol.ResponseHeaderMatchedPriorityLevelConfigurationUID
+	if value := w.Header().Get(key); expectedPL != value {
+		t.Fatalf("expected an HTTP header: %s=%s, but got %s=%s", key, expectedPL, key, value)
+	}
+}
+
+// when a request panics, http2 resets the stream with an INTERNAL_ERROR message
+func expectResetStreamError(t *testing.T, err error) {
+	if err == nil {
+		t.Fatalf("expected the server to send an error, but got nil")
+	}
+
+	uerr, ok := err.(*url.Error)
+	if !ok {
+		t.Fatalf("expected the error to be of type *url.Error, but got: %T", err)
+	}
+	if !strings.Contains(uerr.Error(), "INTERNAL_ERROR") {
+		t.Fatalf("expected a stream reset error, but got: %s", uerr.Error())
+	}
+}
+
+func newClientset(t *testing.T, objects ...runtime.Object) clientset.Interface {
+	clientset := fake.NewSimpleClientset(objects...)
+	if clientset == nil {
+		t.Fatal("unable to create fake client set")
+	}
+	return clientset
+}
+
+// builds a chain of handlers that include the panic recovery and timeout filter, so we can simulate the behavior of
+// a real apiserver.
+// the specified user is added as the authenticated user to the request context.
+func newHandlerChain(t *testing.T, handler http.Handler, filter utilflowcontrol.Interface, userName string, requestTimeout time.Duration) http.Handler {
+	requestInfoFactory := &apirequest.RequestInfoFactory{APIPrefixes: sets.NewString("apis", "api"), GrouplessAPIPrefixes: sets.NewString("api")}
+	longRunningRequestCheck := BasicLongRunningRequestCheck(sets.NewString("watch"), sets.NewString("proxy"))
+
+	handler = WithPriorityAndFairness(handler, longRunningRequestCheck, filter)
+
+	// add the specified user to the request context
+	handler = func(handler http.Handler) http.Handler {
```

Comment:
The only reason that a lambda is constructed and invoked here is to bind `handler` in its body to the value computed at line 509, right?  It would be simpler if line 509 assigned to a variable used only to hold that particular value, then the complexity of contstructing and invoking a lambda would not be needed.